benchmarks/README.md

```diff
@@ -0,0 +1,188 @@
+
+```bash
+python -m benchmarks
+```
+
+The benchmarking module relies on callables that encapsulate the usage of BayBE code in general. The callables are defined in the domains folder, where each domain and every benchmark case has its own file. This file contains the complete definition and all components of a benchmark and is the place where benchmark-related objects, as well as BayBE-related code, are defined. The following components are relevant for a benchmark:
+
+## Benchmarking Structure
+
+The following describes the structure and logic of the benchmarking module, followed by an example of how to create a benchmark.
+
+### Benchmark
+
+```mermaid
+    class Benchmark {
+        <<Generic, BenchmarkSerialization>>
+        +BenchmarkSettingsType settings
+        +Callable function
+        +str name
+        +float|None best_possible_result
+        +list<dict<str, Any>>|None optimal_function_inputs
+        +str description
+        +Result __call__()
+    }
+```
+
+The final benchmark object used for execution holds all relevant information based on the benchmark, covering human-readable information about the domain and the actual benchmarking function. It can be parameterized by the `BenchmarkSettingsType`, which is a generic type defined during runtime and is limited to being a subtype of `BenchmarkSettings` to ensure it covers the minimum necessary information within the added settings subtype class. The `Callable` object is the actual function that performs the benchmarking and contains the BayBE code. It receives an object of the `BenchmarkSettingsType` as input, meaning the input type for the provided function must match the type used for `BenchmarkSettingsType` during the respective benchmark object's creation. The name serves as the unique identifier of the benchmark, and the description is intended to be human-understandable. Note that this identifier is also used to store the results; therefore, any change will be considered a new benchmark. The `best_possible_result` and `optimal_function_inputs` are optional and can provide additional information about the benchmark's domain. The `best_possible_result` is the best achievable result based on the optimization problem, depending on the optimization goal, often referred to as the global minimum or maximum. The `optimal_function_inputs` are a list of input vectors that result in the `best_possible_result` when provided to the lookup function or selected from the DataFrame of the optimization process. The description is a human-readable representation of the benchmark's task, which can provide additional information without restrictions or coupling based on result persistence. Currently, the description is automatically generated based on the `__doc__` attribute of the `function` callable. The `__call__` method is used to execute the benchmark and return the result object. The structure of the result is described under [Result](#result).
```

Comment:
General: Since all of this is a single line it is hard to impossible to given feedback to individual lines of text. For the future, please split such large blocks of text into smaller portions.

Reply:
Also, this large block requires some restructuring: If I am interested in only a single of the arguments, I need to read this whole text block to find it. Please restructure into a list or something more readable.

Reply:
Also, there are a lot of details here that are not necessary to describe in a README:
- "...which is a generic type defined during runtime and is limited to being a subtype of `BenchmarkSettings` to ensure it covers the minimum necessary information within the added settings subtype class." People actually looking at the code will see that, this can be deleted in my opinion.
- "..., meaning the input type for the provided function must match the type used for `BenchmarkSettingsType` during the respective benchmark object's creation.": This basically explains what an input to a function is, so there is no need to describe this.
- "The `best_possible_result` is the best achievable result based on the optimization problem, depending on the optimization goal, often referred to as the global minimum or maximum. The `optimal_function_inputs` are a list of input vectors that result in the `best_possible_result` when provided to the lookup function or selected from the DataFrame of the optimization process.": This explanation is not needed since the argument names exactly describe what they are.